src/Interpreters/Aggregator.cpp

```diff
@@ -250,7 +255,10 @@ size_t Aggregator::Params::getMaxBytesBeforeExternalGroupBy(size_t max_bytes_bef
         }
     }
 
-    return max_bytes_before_external_group_by;
+    if (threshold == std::numeric_limits<size_t>::max())
```

Comment:
This roundabout looks tricky. You can use std::optional instead.

Reply:
Good point, changed to optional